### ISSUE TITLE
Map with a collection value breaks mapper

### ISSUE THREAD

**Issue by larslorenzen** [MorphiaOrg/morphia#1145]
Hi, we recently updated from a very old version of morphia to the 1.3.1.
Now some queries are not working anymore (java.lang.NullPointerException).
While collecting indexes, many classes are added as MappedClass. We also have cases where String.class gets added.

Here an example:
```
public class ReferenceSetQueryTest {

    public static void main(String[] args) {
        MongoClient client = new MongoClient("localhost");
        Morphia morphia = new Morphia();
        morphia.map(A.class, B.class, C.class);

        Datastore datastore = morphia.createDatastore(client, "test");
        datastore.ensureIndexes();

        // create entity A
        A a = new A();
        a.name = "name?";
        datastore.save(a);
        // create entity B with a reference to A
        B b = new B();
        b.as = new HashSet<>();
        b.as.add(a);
        datastore.save(b);

        // this query throws a NullPointerException
        Query<B> bQuery = datastore.find(B.class).filter("as", a);
        System.out.println(bQuery.get());
    }

    @Entity("as")
    public static class A {
        @Id
        private ObjectId id;

        private String name;
    }

    @Entity("bs")
    public static class B {

        @Id
        private ObjectId id;

        @Reference
        private Set<A> as;
    }

    @Entity("cs")
    public static class C {
        @Id
        private ObjectId id;

        private Map<String, Set<String>> strings;
    }

}
```
The multi valued field 'strings' breaks the mapper because the 'Set.class' is added to the mapped classes.

**Comment by raymondjia** [comment]
I ran into the same problem in my project. 

Essentially ensureIndexes() calls collectIndexes(), which calls collectNestedIndexes(). The later doesn't handle collections correctly. If there is a property that is a collection of built-in types (e.g. List<String>), then mapper.getMappedClass() will try to get a mapped class for this built-in type, thus adding it automatically to Mapper.mappedClasses.

Having non-entity classes (such as String, List, Map etc) in Mapper.mappedClasses would later break EmbeddedMapper.readCollection(). 

```
EphemeralMappedField ephemeralMappedField = !mapper.isMapped(mf.getType()) && isMapOrCollection(mf)
                                                            && (mf.getSubType() instanceof ParameterizedType)
                                                        ? new EphemeralMappedField((ParameterizedType) mf.getSubType(), mf, mapper)
                                                        : null;
```

If `mf` is a map or collection, and it is mapped, then ephemeralMappedField won't be generated for the elements of this map or collection.

**Comment by evanchooly** [comment]
The test case above actually works on master so perhaps it was fixed along the way.  If you'd like to confirm that on your end to make sure I haven't mangled your test case, that'd be great.  If not I'll close this in a few days as fixed so I can wrap up the 1.5.0 release.
